db/src/Pos/DB/Block/Internal.hs

```diff
@@ -0,0 +1,290 @@
+        Nothing        -> pure Nothing
+        Just (Left e)  -> throwM (DBMalformed e)
+        Just (Right v) -> pure . Just . Serialized $ uncurry BS.append v
+
+dbPutSerBlundsPureDefault ::
+       forall ctx m. (MonadPureDB ctx m, MonadDB m)
+    => NonEmpty (CB.BlockHeader, SerializedBlund)
+    -> m ()
+dbPutSerBlundsPureDefault (toList -> blunds) = do
+    forM_ blunds $ \(bh, serBlund) -> do
+        (var :: DBPureVar) <- view (lensOf @DBPureVar)
+        flip atomicModifyIORefPure var $
+            (pureBlocksStorage . at (headerHash bh) .~ Just (unSerialized serBlund))
+    putHeadersIndex $ map fst blunds
+
+----------------------------------------------------------------------------
+-- Rocks implementation
+----------------------------------------------------------------------------
+
+-- instance MonadBlockDBGeneric Block
+
+type BlockDBGenericEnv ctx m =
+    ( MonadDBRead m
+    , MonadRealDB ctx m
+    )
+
+dbGetSerBlockRealDefault ::
+       forall ctx m. (BlockDBGenericEnv ctx m)
+    => HeaderHash
+    -> m (Maybe SerializedBlock)
+dbGetSerBlockRealDefault = getSerializedBlock
+
+dbGetSerUndoRealDefault ::
+       forall ctx m. BlockDBGenericEnv ctx m
+    => HeaderHash
+    -> m (Maybe SerializedUndo)
+dbGetSerUndoRealDefault = getSerializedUndo
+
+dbGetSerBlundRealDefault ::
+       forall ctx m. (BlockDBGenericEnv ctx m)
+    => HeaderHash
+    -> m (Maybe SerializedBlund)
+dbGetSerBlundRealDefault hh =
+    Serialized . uncurry BS.append <<$>> getSerializedBlund hh
+
+dbPutSerBlundsRealDefault ::
+       (MonadDB m, MonadRealDB ctx m)
+    => NonEmpty (CB.BlockHeader, SerializedBlund)
+    -> m ()
+dbPutSerBlundsRealDefault = putSerializedBlunds
+
+----------------------------------------------------------------------------
+-- Helpers
+----------------------------------------------------------------------------
+
+getRawData ::  forall m . (MonadIO m, MonadCatch m) => FilePath -> m (Maybe ByteString)
+getRawData  = handle handler . fmap Just . liftIO . BS.readFile
+  where
+    handler :: IOError -> m (Maybe x)
+    handler e
+        | isDoesNotExistError e = pure Nothing
+        | otherwise = throwM e
+
+putRawData ::  MonadIO m => FilePath -> ByteString -> m ()
+putRawData fp v = liftIO $
+    bracket (openBinaryFile fp WriteMode) hClose $ \h ->
+        BS.hPut h v >> hFlush h
+
+deleteData :: (MonadIO m, MonadCatch m) => FilePath -> m ()
+deleteData fp = (liftIO $ removeFile fp) `catch` handler
+  where
+    handler e
+        | isDoesNotExistError e = pure ()
+        | otherwise = throwM e
+
+-- | Paths at which we store the block data.
+data BlockStoragePaths = BlockStoragePaths
+  { bspRoot  :: FilePath
+    -- | Block data itself.
+  , bspBlock :: FilePath
+    -- | Undo information for a block.
+  , bspUndo  :: FilePath
+    -- | Combined storage format. Either this or a combination of 'Block' and
+    -- 'Undo' files will be present.
+  , bspBlund :: FilePath
+  }
```

Comment:
4 space indent

Reply:
You're right. That was code moved from another file :)